pkg/splunk/enterprise/indexercluster.go

```diff
@@ -154,6 +154,13 @@ func ApplyIndexerCluster(client splcommon.ControllerClient, cr *enterprisev1.Ind
 		cr.Status.IdxcPasswordChangedSecrets = make(map[string]bool)
 
 		result.Requeue = false
+		// Set indexer cluster CR as owner reference for clustermaster
+		namespacedName = types.NamespacedName{Namespace: cr.GetNamespace(), Name: GetSplunkStatefulsetName(SplunkClusterMaster, cr.Spec.ClusterMasterRef.Name)}
```

Comment:
Why don't we set the owner reference of ClusterMaster CR vs the ClusterMaster statefulSet? Not sure how its working now.

I thought we are adding ClusterMaster as a watch type & if the ClusterMaster CR is owned by the indexercluster CR we reconcile the indexercluster CR on a change in the Clustermaster CR.

Reply:
I think any change in CM CR is eventually a change in CM statefulset which will trigger the reconcile of indxc. So, I think it should be ok, right? Or am I missing something basic here?

Reply:
If we are adding the IDXC CR as an OwnerReference to the **CM statefulSet** then we don't need to add the ClusterMaster watch type for IDXC CR as it is already watching all statefulSets(by the virtue of statefulSet as a watchtype) and will get reconciled for any changes in a StatefulSet owned by it. 

A change in the CM CR doesn't necessarily lead to a change in the CM statefulSet? Inverse of which might be true as well i.e there can be a change in CM statefulSet without a change in the CM CR. 

IMO we should review both scenarios and do either or both of these:

1. Add CM CR as a watch type and add IDXC CR as an OwnerReference to the CM CR
2. Add IDXC CR as an owner reference to the CM CR's statefulSet.

For the LM ref case, point 1 above should suffice.

Reply:
As discussed offline, we are going with option 2, as we cannot set a CR as owner ref unless it is a default k8s resoruce.